apps/bucket/cloudformation.yml

```
@@ -48,13 +49,57 @@ Resources:
                 - sqs
                 - lambda
     Type: AWS::SNS::TopicPolicy
+
+  LogBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: True
+        IgnorePublicAcls: True
+        BlockPublicPolicy: True
+        RestrictPublicBuckets: True
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+            BucketKeyEnabled: true
+      OwnershipControls:
+        Rules:
+          - ObjectOwnership: BucketOwnerEnforced
+      LifecycleConfiguration:
+        Rules:
+        - Id: ExpirationRule
+          Status: Enabled
+          ExpirationInDays: 30
+
+  LogBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref LogBucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: logging.s3.amazonaws.com
+            Action: s3:PutObject
+            Resource: !Sub "${LogBucket.Arn}/*"
+            Condition:
+              ArnLike:
+                "aws:SourceArn": !GetAtt DataSetBucket.Arn
+              StringEquals:
+                "aws:SourceAccount": !Ref AWS::AccountId
+
   DataSetBucket:
     DeletionPolicy: Retain
     UpdateReplacePolicy: Delete
     DependsOn:
       - SNSTopicPolicy
     Properties:
       BucketName: !Ref DataSetName
+      LoggingConfiguration:
+        DestinationBucketName: !Ref LogBucket
+        LogFilePrefix: s3-access-logs/dataset-bucket/
```

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-loggingconfiguration.html

```suggestion
        TargetObjectKeyFormat:
          PartitionedPrefix:
            PartitionDateSource: EventTime
```